src/hooks/Guilds/ether-swr/useEtherSWR.ts

```diff
@@ -0,0 +1,41 @@
+import { useWeb3React } from '@web3-react/core';
+import useEtherSWRHook, {
+  etherKeyFuncInterface,
+  ethKeyInterface,
+  ethKeysInterface,
+  EthSWRConfigInterface,
+} from 'ether-swr';
+import { useEffect } from 'react';
+import { SWRResponse } from 'swr';
+import { usePrevious } from '../usePrevious';
+
+function useEtherSWR<Data = any, Error = any>(
+  key: ethKeyInterface | ethKeysInterface | etherKeyFuncInterface
+): SWRResponse<Data, Error>;
+function useEtherSWR<Data = any, Error = any>(
+  key: ethKeyInterface | ethKeysInterface | etherKeyFuncInterface,
+  config?: EthSWRConfigInterface<Data, Error>
+): SWRResponse<Data, Error>;
+function useEtherSWR<Data = any, Error = any>(
+  key: ethKeyInterface | ethKeysInterface | etherKeyFuncInterface,
+  fetcher?: any,
+  config?: EthSWRConfigInterface<Data, Error>

```

Comment:
can you fix all the type any's in another issue if you can't do it in this one

Reply:
I copied these TS definitions over from Ether-SWR [here](https://github.com/aboutlo/ether-swr/blob/master/src/useEtherSWR.ts#L35) and they have them set to `any` with the actual types commented as well. Wasn't sure what the reason was, so decided to leave them as is, but will fix them.

Reply:
BTW `Data` and `Error` is intentionally `any` by default, because the users are supposed to override them to whatever type they should actually be. Both SWR and Ether-swr uses `any` as the fallback, so will keep it as is.